core/loader/utils.ts

```diff
@@ -5,14 +5,67 @@ export function isString(attr: null | string): attr is string {
   return typeof attr === 'string' && attr.length > 0
 }
 
+type UrlSegmentType = 'relative' | 'absolute' | 'explicit'
```

Comment:
What do you think of using TypeScript `enum` instead?

The only downside, is `enum`’s is hard to debug in JS env, but it should not be a problem for internal code.

Reply:
Agreed! 

Wanted to go with `enum` here but did a quick search in the entire repo and no results were found so I went with this.
Will update :+1:

Reply:
Feel free to fix another cases when we can use `enum`

Reply:
Do you have a preference from the code style perspective for `enum`s?

Does this look ok? Or you want them in all caps snake_case?

```typescript 
enum UrlSegmentType {
  Absolute = 'absolute',
  Explicit = 'explicit',
  Relative = 'relative'
}
```

Reply:
Looks good. Just let’s try to make identification a little bit shorter. `UrlSegmentType` → `UrlType`